framework/src/media/InputHandler.cpp

```diff
@@ -0,0 +1,342 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include "InputHandler.h"
+#include "MediaPlayerImpl.h"
+
+namespace media {
+namespace stream {
+
+InputHandler::InputHandler()
+	: mInputDataSource(nullptr), mDecoder(nullptr), mIsWorkerAlive(false), mState(BUFFER_STATE_EMPTY), mTotalBytes(0)
+{
+}
+
+void InputHandler::setInputDataSource(std::shared_ptr<InputDataSource> source)
+{
+	mInputDataSource = source;
+}
+
+bool InputHandler::open()
+{
+	if (!getStreamBuffer()) {
+		auto streamBuffer = StreamBuffer::Builder()
+								.setBufferSize(4096)
```

Comment:
These are very important value, please use KConfig or #define instead of constant.